lib/stdlib/src/erl_tar.erl

```diff
@@ -354,7 +361,13 @@ open1({file, Fd}=Handle, read, [raw], Opts) ->
     end;
 open1({file, _Fd}=Handle, read, [], _Opts) ->
     {error, {Handle, {incompatible_option, cooked}}};
-open1(Name, Access, Raw, Opts) when is_list(Name) or is_binary(Name) ->
+open1(Name, Access, Raw, Opts0) when is_list(Name) or is_binary(Name) ->
+    Opts = case lists:member(compressed, Opts0) and lists:member(read, Opts0) of
```

Comment:
```suggestion
open1(Name, Access, Raw, Opts0) when is_list(Name); is_binary(Name) ->
    Opts = case lists:member(compressed, Opts0) andalso lists:member(read, Opts0) of
```

Reply:
No worries, looks like I forgot which language I was writing for!